homeassistant/components/switch/fritzdect.py

```diff
@@ -120,17 +122,61 @@ def is_on(self):
 
     def turn_on(self, **kwargs):
         """Turn the switch on."""
-        actor = self.data.fritz.get_actor_by_ain(self.data.ain)
-        actor.switch_on()
+        if self.data.is_online is False:
+            _LOGGER.error("turn_on: Not online skipping request")
+            return
+
+        try:
+            actor = self.data.fritz.get_actor_by_ain(self.data.ain)
+        except (RequestException, HTTPError):
+            _LOGGER.error("Fritz!Box query failed, triggering relogin")
+            self.data.is_online = False
+
+        try:
+            actor.switch_on()
+        except (RequestException, HTTPError):
+            _LOGGER.error("Fritz!Box query failed, triggering relogin")
+            self.data.is_online = False
```

Comment:
The code looks very similary. I think you should move this two request into same try block